### ISSUE TITLE
New .NET 5.0 Isolated Azure Functions project does not build

### ISSUE THREAD

**Issue by jurgen-ava** [Azure/azure-functions-dotnet-worker#526]
I am using the latest version of Visual Studio 19 (16.10.3).

I also made sure to update the packages to the latest versions (no pending updates in NuGet package manager).

.NET 5.0 Runtime is installed already. I have a DotNetCore 3.1 Azure Function project in the same solution which is working properly.

The new project does not build.

The errors are as follows:

![image](https://user-images.githubusercontent.com/68992734/124594978-b7fd7580-de60-11eb-8da6-68c06048d991.png)

This is how the function code looks like (haven't changed anything yet):

![image](https://user-images.githubusercontent.com/68992734/124595056-d4011700-de60-11eb-8064-7690c16bfb66.png)

The Intellisense suggestions refer to installing WebJobs related packages. I saw on another issue in here, that this is actually wrong for the .NET 5.0 version.

Your assistance would be kindly appreciated.

**Comment by iscottb122** [comment]
@jurgen-ava

It was a while back, but I think I had a similar problem with the default generated function project not being up-to-date for the out-of-process (worker) model. Make sure your csproj/fsproj includes all of:

    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.3.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.0.13" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.0.3" />

(I don't work for MS, I just saw your issue when looking for something else.)

**Comment by iscottb122** [comment]
In fact, there are a few tricky bits to getting started, and the documentation is scattered throughout numerous mysterious issues, so to save you time the csproj/fsproj should look something like this:

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <AssemblyName>Whatever</AssemblyName>
    <AzureFunctionsVersion>v3</AzureFunctionsVersion>
    <_FunctionsSkipCleanOutput>True</_FunctionsSkipCleanOutput>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.3.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.0.13" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.0.3" />
  </ItemGroup>
  <ItemGroup>
    <None Include="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Include="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
</Project>
```

Note the `<_FunctionsSkipCleanOutput>` element and the `<None Include=...>` elements instead of `<None Update=...>`.

Finally, the `FUNCTIONS_WORKER_RUNTIME` is `dotnet-isolated`, so a `local.settings.json` might look something like this:

```
{
  "IsEncrypted": false,
  "Values": {
    "AzureWebJobsStorage": "UseDevelopmentStorage=true",
    "FUNCTIONS_WORKER_RUNTIME": "dotnet-isolated"
  }
}
```

**Comment by brettsam** [comment]
@jurgen-ava -- it looks like you're not referencing the correct packages. The recommendations from @iscottb122 look correct. Did that help get you unblocked?

**Comment by jurgen-ava** [comment]
Hi @iscottb122, many thanks for taking your time to get back to me. Your replies were most helpful and I can now build with no errors.

@brettsam I expected this to work outright as it was a new project created from the latest version of Visual Studio 19. The problem was very likely this missing package: Microsoft.Azure.Functions.Worker.Extensions.Http.
